components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3675,6 +3689,24 @@ public void updateApplication(Application application) throws APIManagementExcep
         APIUtil.sendNotification(applicationEvent, APIConstants.NotifierType.APPLICATION.name());
     }
 
+    private String processGroupIds(String existing, String updated) {
```

Comment:
You need to find the newly added groupIds?
Shall we please add a method comment? Otherwise we always have to fully go through the code to understand what it does as the method name also doesn't give a proper idea.